sourcegraph/cody
# One box: Add search like code snippets

Part of [AI-205](https://linear.app/sourcegraph/issue/AI-205/show-code-snippets-for-context-files)

This PR 
- Adds a search file content code snippets block; it later will be used for search results within chat in one-box mode.
- Adds preview button to context item link which renders highlighted code snippet block (originally this PR was only about code snippet for one box, but I've found having code snippet preview for link very useful, links are usually very long and you don't remember which file it was, helped me to avoid opening a lot of tabs by checking just preview blocks)

<img width="1136" alt="Screenshot 2024-09-11 at 00 45 07" src="https://github.com/user-attachments/assets/6c21c8b1-183b-43e0-8c71-d6ee21beec1c">

## Todo
- [x] Add an unhighlighted text block 
- [x] Add highlights support 
- [x] Add demo 
- [x] Align colors for different vscode themes 

## Test plan
- CI is green
- Check storybook demo 
- Run VSCode (or Cody Web) check that you can see preview code blocks next to file context files 



## Changelog
